WalletWasabi/WabiSabi/Backend/Banning/CoinVerifier.cs

```diff
@@ -158,18 +158,20 @@ private void CleanUp()
 		}
 
 		bool shouldBan = flagIds.Any(id => WabiSabiConfig.RiskFlags.Contains(id));
-		bool shouldRemove = shouldBan || !response.Report_info_section.Address_used;
+
+		// When to remove: if we ban it OR if address_used is false, API provider doesn't know about it OR if the report doesn't include the block in which the script is used to receive the coin.
+		bool shouldRemove = shouldBan || !response.Report_info_section.Address_used || blockchainHeight - (confirmations - 1) > response.Report_info_section.Report_block_height;
```

Comment:
I'm not quite sure with this.
The road of nullabilities is long and deep. At this point, if `shouldBan is false`,`address_used is true`, but either `confirmations` or `blockchainHeight` is `null`, `shouldRemove` will be false.
`null - 1` is null, `null > int` is null.
We should add a sanity check here. In case of missing information we should remove at least?

Reply:
> The road of nullabilities is long and deep. At this point, if shouldBan is false,address_used is true, but either confirmations or blockchainHeight is null, shouldRemove will be false.

I don't get this. They are all separated with OR `||`, if one of them is `true`, we will remove it.

I will try to make this without nullables. if I cannot, I will add some sanity checks.

Reply:
Yes, my example with `address_used` as `true` might be misleading, as we negate that value, so at comparison it will count as `false` actually.

What I'm saying if I understand it correctly:
- We don't ban it (`shouldBan = false`)
- The address was processed (`address_used = true`)
- It was confirmed 4 times (confirmations = 4)
- But for example, we didn't get the blockchainHeight (blockchainHeight = null).
Then it's
```
bool shouldRemove = false || !true || null - 3 > response.Report_info_section.Report_block_height;
```
which will be `false` at the end, so we don't remove it, even thought we can't be 200% sure it's safe.
Does it make sense?

Reply:
Yes, I misunderstood your example.

I got what you mean. 
The only problem is that I broke every existing test possible, when I tried to remove the nullability. 

I will come up with something.

Reply:
None of those parameter should be null. I think they are null because they were put at the end of the parameters list after another parameter that is optional and that is also null for the sake of nullability. It would be good to improve a bit that code or at least that method.

Reply:
Removed nullability in 6121a24